### ISSUE TITLE
Add specific microarchitecture support for AArch64 processors

### ISSUE THREAD

**Issue by alalazo** [spack/spack#13481]
As a user I would like to have microarchitecture specific support for processors in the `AArch64` family so that I can compile code optimized for them similarly to what is already possible for Intel, AMD or Power CPUs.
 
### Rationale

The idea is to add more specific targets in the `AArch64` family.

### Description

Due to the model adopted by ARM, where a CPU can be identified by a base model with some additional features added or removed, it might be possible that we need to generalize the structure currently used in `microarchitectures.json` and/or adapt how we pass flags to compilers.

### Additional information

A list of references to get information on `AArch64` and compiler flags follows:

- [ARM official documentation](https://developer.arm.com/docs)
- [GCC online documentation](https://gcc.gnu.org/onlinedocs/)
- [LLC online documentation](https://llvm.org/docs/CommandGuide/llc.html)

For LLVM the way to get registered targets for the assembler is:
```console
$  llc-9 --version
LLVM (http://llvm.org/):
  LLVM version 9.0.1
  
  Optimized build.
  Default target: x86_64-pc-linux-gnu
  Host CPU: broadwell

  Registered Targets:
    aarch64    - AArch64 (little endian)
    aarch64_32 - AArch64 (little endian ILP32)
    aarch64_be - AArch64 (big endian)
    amdgcn     - AMD GCN GPUs
    arm        - ARM

[ ... ]

$ llc-9 --march=aarch64 -mcpu=help
Available CPUs for this target:

  apple-latest - Select the apple-latest processor.
  cortex-a35   - Select the cortex-a35 processor.
  cortex-a53   - Select the cortex-a53 processor.
  cortex-a55   - Select the cortex-a55 processor.
  cortex-a57   - Select the cortex-a57 processor.
  cortex-a72   - Select the cortex-a72 processor.
  cortex-a73   - Select the cortex-a73 processor.
  cortex-a75   - Select the cortex-a75 processor.
  cortex-a76   - Select the cortex-a76 processor.

[ ... ]
```

**Comment by alalazo** [comment]
@becker33 @scheibelp @tldahlgren Currently we support `thunderx2` and `a64fx`. Which other `aarch64` microarchitecture do we need to include to consider this issue closed for v0.15.0?

**Comment by alalazo** [comment]
@tgamblin @becker33 Moved this to the next minor release. It would be good if after v0.15.0 we could define a minimal list of uarchs that is necessary to close this.
